R/utilities.R

```diff
@@ -306,27 +324,29 @@ throwWarningAboutDifferentStatistics <- function(multiple.statistics, function.n
 
 # No doubt there is a better way of storing this than a function
 # This vector contains common names for statistics (in Q, Displayr, and elsewhere)
-statisticNames <- function()   c('%', '% column responses', '% column share',
-                                 '% excluding nan', '% responses', '% row responses',
-                                 '% row share', '% share', '% total responses', '% total share',
-                                 '25th percentile', '5th percentile', '75th percentile', '95th percentile',
-                                 'average', 'base', 'coefficient', 'column %', 'column comparisons',
-                                 'column names', 'column sample size', 'column standard error',
-                                 'column standard error of mean', 'columns compared', 'corrected p',
-                                 'correlation', 'count', 'cumulative %', 'd.f.', 'degrees of freedom',
-                                 'df', 'df', 'effective column sample size', 'effective count',
-                                 'effective row sample size', 'effective sample size',
-                                 'expected average', 'expected correlation', 'expected n',
-                                 'expected percent', 'expectedpercent', 'index', 'interquartile range',
-                                 'interquartilerange', 'labels', 'maximum', 'mean', 'median',
-                                 'minimum', 'missing count', 'mode', 'mode',
-                                 'multiple comparison adjustment', 'n observations', 'net',
-                                 'not duplicate', 'observation', 'percent', 'probability %',
-                                 'range', 'row %', 'row sample size', 's.e.', 'sample size',
-                                 'sample size', 'se', 'standard deviation', 'standard error',
-                                 'sum', 't-statistic', 'text', 'total', 'total %', 'trimmed average',
-                                 'values', 'weighted column sample size', 'weighted count',
-                                 'weighted row sample size', 'weighted sample size', 'z', 'z-statistic')
+statisticNames <- function()   c("%", "% column responses", "% column share", "% excluding nan",
```

Comment:
Is this still needed given what we discussed about just checking that the input is a Q Table and then we know what the stat names are?

Reply:
Need to distinguish between a dim created via a span and a dimension created via multiple stats. The last array dimnames could be inspected and checked against the span attribute possibly and if no match then it must be generated from stats. I haven't checked if that breaks other situations yet. It could be refactored for that logic though. The goal is to preserve the last dim if there are multiple stats (not flatten that dim) so a 3, 4 or 5d array is flattened to a 3d array if multiple stats are present.

Reply:
The vector is also used to check if a 1d or 2d array is comprised of multiple stats too. I don't think this vector can be completely removed unless they're is another way to know the table has multiple stats in another way